PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx

```diff
@@ -150,17 +156,20 @@ struct HfTaskFlowCharmHadrons {
     float pYtrack1 = cand.pyProng1();
     float pTtrack1 = cand.ptProng1();
     float phiTrack1 = std::atan2(pYtrack1, pXtrack1);
-    float pXtrack2 = cand.pxProng2();
-    float pYtrack2 = cand.pyProng2();
-    float pTtrack2 = cand.ptProng2();
-    float phiTrack2 = std::atan2(pYtrack2, pXtrack2);
 
     tracksQx.push_back(cos(harmonic * phiTrack0) * pTtrack0 / ampl);
     tracksQy.push_back(sin(harmonic * phiTrack0) * pTtrack0 / ampl);
     tracksQx.push_back(cos(harmonic * phiTrack1) * pTtrack1 / ampl);
     tracksQy.push_back(sin(harmonic * phiTrack1) * pTtrack1 / ampl);
-    tracksQx.push_back(cos(harmonic * phiTrack2) * pTtrack2 / ampl);
-    tracksQy.push_back(sin(harmonic * phiTrack2) * pTtrack2 / ampl);
+
+    if constexpr (DecayChannel != DecayChannel::D0ToPiK) {
```

Comment:
Please avoid this kind of confusing naming patterns. `DecayChannel` on the left is a variable, `DecayChannel` on the right is a type.

Reply:
Sorry, I have closed this PR due to a failed submission and will resubmit. I'll change or explain existing comments.